src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        try {
+            return HibernateUtil.getFactory().fromTransaction(session -> {
+                session.persist(movieSession);
+                return movieSession;
+            });
```

Comment:
Maybe inTransaction we need to use or it works in similar way? I confused because in another DAO you use inTransaction, so if it’s the same, then it’s better to write in the same style

Reply:
Almost the same. But maybe you are right. It would be better to write in the same style =)